build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/transportversion/BumpTransportVersionTask.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.transportversion;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class BumpTransportVersionTask extends DefaultTask {
+
+    @TaskAction
+    public void run() throws IOException {
+        Path transportVersionFile = Path.of("./server/src/main/java/org/elasticsearch/TransportVersion.java");
+        String content = Files.readString(transportVersionFile);
+
+        Pattern nextVersionPattern = Pattern.compile("// NEXT_VERSION ([\\d_]+)");
```

Comment:
Rather than using a fixed comment in the file, can't you parse for the CURRENT version and extract it based on that?

Reply:
In the release automation, we look backwards from the end of the file, finding the first occurrence of the pattern `public static final TransportVersion V_[\d_]+ = `. I think we could use the same idea here, finding the version number from the right side of that equals. Then we don't need a special comment.